test/functional/formatting_stack/core.clj

```diff
@@ -0,0 +1,37 @@
+(ns functional.formatting-stack.core
+  (:require
+   [clojure.test :refer :all]
+   [formatting-stack.core :as sut]
+   [formatting-stack.protocols.formatter :as formatter]
+   [formatting-stack.protocols.reporter :as protocols.reporter]
+   [matcher-combinators.test :refer [match?]]
+   [nedap.utils.modular.api :refer [implement]])
+  (:import (clojure.lang ExceptionInfo)))
+
+(defn throw-exception [& args]
+  (throw (ex-info "Kaboom!" {})))
+
+(def failing-formatter
+  (implement {:id ::failing-formatter}
+    formatter/--format! throw-exception))
+
+(defn store-in-latest [{::keys [latest]} args]
+  (reset! latest args))
+
+(defn recording-recorder [latest]
+  (implement {::latest latest}
+    protocols.reporter/--report store-in-latest))
+
+(deftest format!
+  (testing "exceptions are passed to the reporter"
+    (let [x (atom nil)]
+      (sut/format! :formatters [failing-formatter]
+                   :linters []
+                   :processors []
+                   :reporter (recording-recorder x)
+                   :in-background? false)
+      (is (match? [{:source    :formatting-stack/process!
```

Comment:
`u.t/expect`?

It would _look_ like overkill atm, but one never knows how tests could grow

Reply:
I tried `u.t/expect`, but it's a bit hard to use in this case, as it uses `meta=`. Kept the test as is, can reevaluate when we expand it.